infra/storage/kFile.class.php

```diff
@@ -722,6 +722,23 @@ public static function safeFilePutContents($filePath, $var, $mode=null)
 		}
 		return true;
 	}
+
+	/**
+	 * Lazy saving of file content to a temporary path, the file will exist in this location until the temp files are purged
+	 * @param string $fileContent
+	 * @param string $prefix
+	 * @param integer $permission
+	 * @return string path to temporary file location
+	 */
+	public static function createTempFile($fileContent, $prefix = '' , $permission = null)
+	{
+		$tempDirectory = sys_get_temp_dir();
+		$fileLocation = tempnam($tempDirectory, $prefix);
+		file_put_contents($fileLocation, $fileContent);
+		if ($permission)
+			chmod($fileLocation, $permission);
```

Comment:
All file system access function that are touch , only via kFile.
in this case kFile::chmod